force-app/main/default/classes/DMLOperations.cls

```diff
@@ -161,14 +182,58 @@ public with sharing class DMLOperations {
 	public static Account upsertAccount(String accountName) {
 		// Query for the Account
 		List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
+        Account acct = new Account();
+		if (!accounts.isempty()){
+            acct = accounts[0];
+            acct.Description = 'Updated Account';   // if the Account exists, update the description
+        } else {                                    // otherwise, create a new Account and set the description
+            acct.Name = accountName;
+            acct.Description = 'New Account';
+        }   
+		upsert acct;                        		// Upsert the Account
+		return acct;
+	}
+    
+    /**
+     * Utility 9.5
+     * Created to facilitate Question 10
+	 * Inserts a list of Accounts with specified names.
+	 * For each account in the list:
+        * If the Account with the given name already exists, update the description to "Updated Account".
+	    * If no such Account exists, it creates a new one and sets the description to "New Account".
+	 * @param accountName The name of the Account to be upserted.
+	 * @return The Account that was upserted.
+	 */
+	
+    public static Map<String, Id> insertAccounts(Set<String> acctNamesSet) {
+		// Query for the Account
+        if (acctNamesSet.isEmpty()){
+            return null;
+        }
+        Map<String, Id> nameToIdMap = new Map<String, Id>();
+        List<Account> accounts = new List<Account>([SELECT Id, Name FROM Account WHERE Name IN :acctNamesSet]);
+		if (!accounts.isEmpty()){                   //if Accounts exist...
+            for(Account acct : accounts){               // iterate through the Accounts
+                acctNamesSet.remove(acct.Name);         // remove the Account's name from Set
+                nameToIdMap.put(acct.Name, acct.Id);    // add Account's Name & Id to Map
+            }
+        } 
+        accounts.clear();                           // clears the list
+        for (String acctName: acctNamesSet){        // Iterate through (remaining) Set of Names, for each name...
+            Account newAccount = new Account(       // create a new Account, setting the Name and Description fields
+                Name = acctName);
+            accounts.add(newAccount);               //Add new account to the List -- accounts            
+        } 
+        insert accounts;                        	// Insert the new Accounts
+        for (Account acct : accounts){
+            nameToIdMap.put(acct.Name, acct.Id);
+        }
+		return nameToIdMap;
+	}
+    
```

Comment:
You applied the same remove pattern you used above and I think it was a little more verbose here. If you could have put the Account right into your map, you could have done an `insert accounts.values()`; instead of having to re-loop through a few times. But I think you get the overall strategy of using bulk handling to avoid the DML in loops